src/core/algorithms/dd/split/split.cpp

```diff
@@ -629,34 +668,35 @@ std::list<DD> Split::HybridPruningReduce(DF const& rhs, std::vector<DF> const& s
     return dds;
 }
 
-std::list<DD> Split::InstanceExclusionReduce(
-        std::vector<std::pair<std::size_t, std::size_t>> const& tuple_pairs,
-        std::vector<DF> const& search, DF const& rhs, unsigned& cnt) {
+std::list<DD> Split::InstanceExclusionReduce(std::vector<std::size_t> const& tuple_pair_indices,
+                                             std::vector<DF> const& search, DF const& rhs,
+                                             unsigned& cnt) {
     if (!search.size()) return {};
 
     std::list<DD> dds;
     DF const first_df = search.front();
     DF const last_df = search.back();
-    std::vector<std::pair<std::size_t, std::size_t>> remaining_tuple_pairs;
+    std::vector<std::size_t> remaining_tuple_pair_indices;
 
     cnt++;
     bool last_dd_holds = true;
     bool no_pairs_left = true;
-    for (auto pair : tuple_pairs) {
-        if (!CheckDF(rhs, pair)) {
-            if (CheckDF(first_df, pair)) {
-                remaining_tuple_pairs.push_back(pair);
+    for (auto index : tuple_pair_indices) {
+        if (!CheckDF(rhs, tuple_pairs_[index])) {
+            if (CheckDF(first_df, tuple_pairs_[index])) {
+                remaining_tuple_pair_indices.push_back(index);
                 no_pairs_left = false;
             }
-            if (last_dd_holds && CheckDF(last_df, pair)) last_dd_holds = false;
+            if (last_dd_holds && CheckDF(last_df, tuple_pairs_[index])) last_dd_holds = false;
             if (!no_pairs_left && !last_dd_holds) break;
         }
     }
 
     if (no_pairs_left) {
         if (IsFeasible(first_df)) dds.emplace_back(first_df, rhs);
         std::vector<DF> remainder = DoPositivePruning(search, first_df);
-        std::list<DD> remaining_dds = InstanceExclusionReduce(tuple_pairs, remainder, rhs, cnt);
+        std::list<DD> remaining_dds =
+                InstanceExclusionReduce(tuple_pair_indices, remainder, rhs, cnt);
         dds.splice(dds.end(), remaining_dds);
```

Comment:
maybe
```suggestion
        dds.splice(dds.end(), std::move(remaining_dds));
```
?

We don't need `remaining_dds` anymore